src/screens/VisualTests/BuildProgress.tsx

```diff
@@ -20,46 +38,90 @@ export const Bar = styled.div<{ percentage: number }>(({ theme, percentage }) =>
   height: "100%",
   left: "0",
   width: `${percentage}%`,
-  transition: "all 150ms ease-out",
+  transition: "width 3s ease-out",
   backgroundColor: theme.background.hoverable,
 }));
 
-export const Text = styled.div({
-  position: "relative",
-  zIndex: 1,
+const ExpandableDiv = styled.div<{ expanded: boolean }>(({ expanded, theme }) => ({
+  display: "grid",
+  gridTemplateRows: expanded ? "1fr" : "0fr",
+  background: theme.background.app,
+  borderBottom: expanded ? `1px solid ${theme.appBorderColor}` : "none",
+  transition: "grid-template-rows 150ms ease-out",
+}));
+
+const StepDetails = styled.div({
+  whiteSpace: "nowrap",
+  overflow: "hidden",
 });
 
+const StepDetail = styled.div<{ isCurrent: boolean; isPending: boolean }>(
+  ({ isCurrent, isPending }) => ({
+    display: "flex",
+    flexDirection: "row",
+    gap: 8,
+    opacity: isPending ? 0.7 : 1,
+    fontWeight: isCurrent ? "bold" : "normal",
+    fontFamily: "Menlo, monospace",
+    fontSize: "12px",
+    lineHeight: "24px",
+    margin: "0 10px",
+    "&:first-of-type": {
+      marginTop: 8,
+    },
+    "&:last-of-type": {
+      marginBottom: 8,
+    },
+  })
+);
+
+const StepIcon = styled(Icons)(
+  { width: 10, marginRight: 8 },
+  ({ icon }) => icon === "sync" && css({ animation: `${spin} 1s linear infinite` })
+);
+
 type BuildProgressProps = {
-  runningBuild?: RunningBuildPayload;
-  switchToNextBuild?: () => void;
+  buildProgress?: RunningBuildPayload;
+  expanded?: boolean;
 };
 
-export function BuildProgress({ runningBuild, switchToNextBuild }: BuildProgressProps) {
-  const percentage = runningBuild?.buildProgressPercentage;
+export function BuildProgress({ buildProgress, expanded }: BuildProgressProps) {
+  const stepHistory = useRef<
+    Partial<Record<RunningBuildPayload["currentStep"], RunningBuildPayload>>
+  >({});
 
-  // eslint-disable-next-line no-nested-ternary
-  const text = runningBuild ? (
-    <BuildProgressLabel runningBuild={runningBuild} />
-  ) : switchToNextBuild ? (
-    "There's a newer snapshot with changes"
-  ) : (
-    "Reviewing is disabled because there's a newer snapshot on <branch>"
-  );
+  useEffect(() => {
+    stepHistory.current[buildProgress.currentStep] = { ...buildProgress };
+  }, [buildProgress]);
```

Comment:
Does this approach work if you change story or close+reopen the addon panel mid-build?

Reply:
Yes, we don't remount the component when those things happen. I checked.